### PR TITLE
Feature/coininfopage

### PR DESCRIPTION
**Pull request by cristhianbenitez**
Coin Information Page created displaying information such as  Price, ATH, ATL, Market Cap, Fully diluted valuation, Volume, Volume/Market, Total Volume, Circulating Supply, Max Supply, Description and Links to websites related with the Coin.
Also created the Coin Currency Converter and a Dropdown menu to select intervals (not functionality yet)

### CODE REVIEW

File: src/Components/AllTimeInfo/AllTimeInfo.js [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+  MiddleContentWrapper,
+  RedArrowDown,
+  TextContainer
+} from './AllTimeInfo.styles';
+
+export const AllTimeInfo = (props) => {
+  const renderRedOrGreenArrow = (price) =>
+    price < 0 ? <RedArrowDown /> : <GreenArrowUp />;
+  var athDate = new Date(props.athDate).toGMTString();
+  var atlDate = new Date(props.atlDate).toGMTString();
```
Comment (yoloOnTheBattlefield) [inline src/Components/AllTimeInfo/AllTimeInfo.js:19]: wtf var

File: src/Components/CurrencyConverter/CurrencyConverter.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+  state = {
+    firstCurrency: localStorage.selection,
+    secondCurrency: this.props.coinSymbol,
+    firstValue: 1,
+    secondValue: 1
+  };
+
+  handleFirstValueOnChange = (value) => {
+    this.setState({ ...this.state, firstValue: value });
+    21;
```
Comment (yoloOnTheBattlefield) [inline src/Components/CurrencyConverter/CurrencyConverter.js:20]: what is 21 supposed to do?

File: src/Components/CurrencyConverter/CurrencyConverter.js [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+  handleSecondValueOnChange = (value) => {
+    this.setState({ ...this.state, secondValue: value });
+  };
+  handleSwap = (state) => {
+    const swappedState = {
+      firstValue: state.secondValue,
+      firstCurrency: state.secondCurrency,
+      secondValue: state.firstValue,
+      secondCurrency: state.firstCurrency
+    };
```
Comment (yoloOnTheBattlefield) [inline src/Components/CurrencyConverter/CurrencyConverter.js:31]: what are you doing here?

File: src/Components/IntervalDropdown/IntervalDropdown.js [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+  };
+
+  toggle = () =>
+    this.setState((prevState) => ({
+      isOpen: !prevState.isOpen
+    }));
+
+  handleOnClick = (e) => {
+    this.setState({ selection: e.target.outerText });
+    this.toggle;
```
Comment (yoloOnTheBattlefield) [inline src/Components/IntervalDropdown/IntervalDropdown.js:28]: and this?

**Review by yoloOnTheBattlefield — CHANGES_REQUESTED** [review]
Why are you not using absolute imports bro?

File: src/Components/CurrencyConverter/CurrencyConverter.js (hunk 2 above)
Comment (cristhianbenitez) [reply src/Components/CurrencyConverter/CurrencyConverter.js:20]: lol, fixed

**Comment by cristhianbenitez** [comment]
> and this? 

trying to target the text content to update the dropdown title, now i used a another approach

**Comment by cristhianbenitez** [comment]
> what are you doing here? 

trying to swap

File: src/Components/ChartsLegend/ChartsLegend.js [hunk 5]
```diff
@@ -1,12 +1,12 @@
+import getSymbolFromCurrency from 'currency-symbol-map';
 import React, { Component } from 'react';
+import { formattedNumber, todayDate } from 'Utils';
```
Comment (yoloOnTheBattlefield) [inline src/Components/ChartsLegend/ChartsLegend.js:3]: new line

File: src/Components/CoinInfo/CoinInfo.styles.js [hunk 6]
```diff
@@ -0,0 +1,65 @@
+import LinkIcon from 'Assets/LinkIcon.png';
+import styled from 'styled-components';
```
Comment (yoloOnTheBattlefield) [inline src/Components/CoinInfo/CoinInfo.styles.js:2]: styled before link

File: src/Components/CoinInfo/CoinInfo.styles.js [hunk 7]
```diff
@@ -0,0 +1,65 @@
...
+    text-decoration: none;
+    color: #ffff;
+  }
+`;
+export const BottomContent = styled.div`
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  padding-top: 1em;
+`;
```
Comment (yoloOnTheBattlefield) [inline src/Components/CoinInfo/CoinInfo.styles.js:65]: you have with some variables a new line between them, some without. pick one

File: src/Components/CoinsListItem/CoinsListItem.js [hunk 8]
```diff
@@ -41,7 +41,8 @@ export class CoinsListItem extends Component {
       } = coinInfo;
       const renderRedOrGreenArrow = (price) =>
         price < 0 ? <RedArrowDown /> : <GreenArrowUp />;
-
+      ('====================================');
+      ('====================================');
```
Comment (yoloOnTheBattlefield) [inline src/Components/CoinsListItem/CoinsListItem.js:45]: do you need this?

File: src/Components/IntervalDropdown/IntervalDropdown.styles.js [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+  display: flex;
+  justify-content: space-between;
+`;
+export const DropDownHeader = styled.div`
+  cursor: pointer;
+  padding-right: 2em;
+  height: 100%;
+`;
+
+export const DropDownList = styled.ul`
```
Comment (yoloOnTheBattlefield) [inline src/Components/IntervalDropdown/IntervalDropdown.styles.js:26]: same here

File: src/Components/SubNavbar/SubNavbar.styles.js [hunk 10]
```diff
@@ -1,6 +1,6 @@
+import Bitcoin from 'Assets/BitcoinIcon.png';
+import Ethereum from 'Assets/EthereumIcon.png';
 import styled from 'styled-components';
```
Comment (yoloOnTheBattlefield) [inline src/Components/SubNavbar/SubNavbar.styles.js:3]: stled before relative imports

File: src/Pages/CoinInformation/CoinInformation.styles.js [hunk 11]
```diff
@@ -0,0 +1,74 @@
+import BackgroundImage from 'Assets/BackgroundImage.png';
+import styled from 'styled-components';
+
+const StyledDiv = styled.div`
+  // General Div
```
Comment (yoloOnTheBattlefield) [inline src/Pages/CoinInformation/CoinInformation.styles.js:5]: no need for a comment

File: src/Pages/CoinInformation/CoinInformation.styles.js [hunk 12]
```diff
@@ -0,0 +1,74 @@
+import BackgroundImage from 'Assets/BackgroundImage.png';
+import styled from 'styled-components';
```
Comment (yoloOnTheBattlefield) [inline src/Pages/CoinInformation/CoinInformation.styles.js:2]: abs before rel

File: src/Pages/Coins/Coins.js [hunk 13]
```diff
@@ -1,13 +1,13 @@
+import coinGecko from 'Api/coinGecko';
+import { Charts, CoinsList } from 'Components';
 import React, { Component } from 'react';
```
Comment (yoloOnTheBattlefield) [inline src/Pages/Coins/Coins.js:3]: absolute before rel

File: src/index.js [hunk 14]
```diff
@@ -1,6 +1,6 @@
+import App from 'App';
```
Comment (yoloOnTheBattlefield) [inline src/index.js:1]: abs before rel

**Review by yoloOnTheBattlefield — CHANGES_REQUESTED** [review]

File: src/Components/CurrencyConverter/CurrencyConverter.js [hunk 15]
```diff
@@ -0,0 +1,82 @@
...
+      currencies: swappedState
+    });
+  };
+  render() {
+    const convertedValue =
+      this.state.currencies[0]?.currencyName !== this.props.coinSymbol
+        ? (
+            this.state.currencies[0]?.currencyValue / this.props.coinPrice
+          )?.toFixed(6)
+        : this.state.currencies[0]?.currencyValue *
```
Comment (yoloOnTheBattlefield) [inline src/Components/CurrencyConverter/CurrencyConverter.js:47]: you can extract currency value in a variable

File: src/Components/Dropdown/Dropdown.styles.js [hunk 16]
```diff
@@ -1,21 +1,24 @@
 import styled from 'styled-components';
-import Currency from '../../Assets/CurrencyIcon.svg';
-import ArrowUp from '../../Assets/ArrowUp.png';
-import ArrowDown from '../../Assets/ArrowDown.png';
+import ArrowDown from 'Assets/ArrowDown.png';
+import ArrowUp from 'Assets/ArrowUp.png';
+import Currency from 'Assets/CurrencyIcon.svg';
```
Comment (yoloOnTheBattlefield) [inline src/Components/Dropdown/Dropdown.styles.js:4]: assets lowercase

File: src/Components/IntervalDropdown/IntervalDropdown.js [hunk 17]
```diff
@@ -0,0 +1,72 @@
...
+      <IntervalDropdownWrapper>
+        <DropDownContainer
+          tabIndex={0}
+          role="button"
+          onKeyPress={() => {
+            this.toggle();
+          }}
+          onClick={() => {
+            this.toggle();
+          }}
```
Comment (yoloOnTheBattlefield) [inline src/Components/IntervalDropdown/IntervalDropdown.js:50]: use short form
onClick={this.toggle}

File: src/Components/IntervalDropdown/IntervalDropdown.styles.js [hunk 18]
```diff
@@ -0,0 +1,86 @@
+import ArrowDown from 'Assets/ArrowDown.png';
+import ArrowUp from 'Assets/ArrowUp.png';
+import styled from 'styled-components';
```
Comment (yoloOnTheBattlefield) [inline src/Components/IntervalDropdown/IntervalDropdown.styles.js:3]: absolute before relative

File: src/Components/LinkContainer/LinkContainer.js [hunk 19]
```diff
@@ -0,0 +1,28 @@
...
+    <CoinLinkContainer extraIcon={extraIcon}>
+      <IconContainer>
+        <ChainIcon extraIcon={extraIcon} />
+      </IconContainer>
+      <SiteLink href={urlLink}>{cleanUrlString(urlLink)}</SiteLink>
+      {extraIcon ? (
+        <IconContainer>
+          <TabsIcon />
+        </IconContainer>
+      ) : null}
```
Comment (yoloOnTheBattlefield) [inline src/Components/LinkContainer/LinkContainer.js:25]: can use logical or `&&`

File: src/Components/LinkContainer/LinkContainer.styles.js [hunk 20]
```diff
@@ -0,0 +1,38 @@
+import LinkIcon from 'Assets/LinkIcon.png';
+import TabsIconImage from 'Assets/TabsIcon.png';
+import styled from 'styled-components';
```
Comment (yoloOnTheBattlefield) [inline src/Components/LinkContainer/LinkContainer.styles.js:3]: abs before rel

File: src/Components/Navbar/Navbar.js [hunk 21]
```diff
@@ -22,7 +20,7 @@ class Navbar extends Component {
       const upperCaseArr = data.map((el) => el.toUpperCase());
       this.setState({ supportedCoins: upperCaseArr });
     } catch (err) {
-      console.log('err');
+      ('err');
```
Comment (yoloOnTheBattlefield) [inline src/Components/Navbar/Navbar.js:23]: huh?

File: src/Helpers/WithRouter.js [hunk 22]
```diff
@@ -1,10 +1,11 @@
 // With Router Higher Order Component
```
Comment (yoloOnTheBattlefield) [inline src/Helpers/WithRouter.js:1]: delete this

**Review by yoloOnTheBattlefield — CHANGES_REQUESTED** [review]

File: src/Components/CoinsListItem/CoinsListItem.js [hunk 23]
```diff
@@ -40,10 +40,10 @@ export class CoinsListItem extends Component {
         sparkline_in_7d: pricesOfLastSevenDays
       } = coinInfo;
 
-      const renderRedOrGreenArrow = (price) =>
+      const conditionalRenderArrow = (price) =>
```
Comment (yoloOnTheBattlefield) [inline src/Components/CoinsListItem/CoinsListItem.js:43]: find a better name

File: src/Components/CoinsListItem/CoinsListItem.js [hunk 24]
```diff
@@ -84,49 +84,67 @@ export class CoinsListItem extends Component {
...
             {displayPositiveNumber(weeklyChanges)}%
           </PriceChangePercentage>
           <TableData>
             <PercentageBarContainer>
               <ValuesContainer>
                 <Value
-                  colors={percBarColors.left[index % percBarColors.left.length]}
+                  colors={
+                    percentageBarColors.left[
+                      index % percentageBarColors.left.length
```
Comment (yoloOnTheBattlefield) [inline src/Components/CoinsListItem/CoinsListItem.js:104]: put this in a variable and reuse it

File: src/Components/CoinsListItem/CoinsListItem.js [hunk 25]
```diff
@@ -136,33 +154,51 @@ export class CoinsListItem extends Component {
...
+                    ]
+                  }
                 >
                   <BulletCircle
                     colors={
-                      percBarColors.left[index % percBarColors.left.length]
+                      percentageBarColors.left[
+                        index % percentageBarColors.left.length
+                      ]
```
Comment (yoloOnTheBattlefield) [inline src/Components/CoinsListItem/CoinsListItem.js:167]: const color =  percentageBarColors.left[ index % percentageBarColors.left.length ] and then you can reuse it. DRY

File: src/Components/IntervalDropdown/IntervalDropdown.js [hunk 26]
```diff
@@ -45,9 +45,7 @@ export class IntervalDropdown extends React.Component {
           onKeyPress={() => {
             this.toggle();
           }}
```
Comment (yoloOnTheBattlefield) [inline src/Components/IntervalDropdown/IntervalDropdown.js:47]: could be in one line

File: src/Components/index.js [hunk 27]
```diff
@@ -1,5 +1,5 @@
 // export all components from here
```
Comment (yoloOnTheBattlefield) [inline src/Components/index.js:1]: no need for obvious comment

**Review by yoloOnTheBattlefield — CHANGES_REQUESTED** [review]

**Review by yoloOnTheBattlefield — APPROVED** [review]

**Pull request merged**